subsys/bluetooth/host/Kconfig

```diff
@@ -84,6 +84,7 @@ config BT_HCI_RESERVE
 	default 1 if BT_ESP32
 	default 0 if BT_B91
 	default 1 if BT_AMBIQ_HCI
+	default 1 if BT_SILABS_HCI
```

Comment:
Is this change actually related to the shell? These values should relate to HCI-driver internal implementation details. If the silabs driver requires an extra byte, that's fine, but I think it would be cleaner to have that as a separate bug fix commit with a proper commit message explaining this (instead of mixing it together with a shell-related change).

Reply:
It was required to make the BLE Shell operationnal on that board yes, I don't in depth knowledge of zephyr/ble to know where it's more practical to have it implemented

Reply:
The one byte headroom requirement seems to come from here in the silabs HCI driver send() implementation:
https://github.com/zephyrproject-rtos/zephyr/blob/868193ef43e90b6f55753d8baa38701ef09ed8d9/drivers/bluetooth/hci/slz_hci.c#L98-L104

So it's an actual bug that indeed needed fixing. To be clear: I was only asking to split up your changes in this pull request to two separate commits, since the buffer headroom requirement has nothing to do with the shell (the bug could manifest itself in any Bluetooth use case with this HCI driver).

Reply:
Ok I'll make a separe PR. How does this work, should I commit and remove the amend to that line? Can the edit be made directly inside the PR ?

Reply:
You could e.g. do it using something like:
```
git reset HEAD~2
git add boards/arm/efr32xg24_dk2601b/Kconfig.defconfig
git commit -v -s # Create the first commit message
git add subsys/bluetooth/host/Kconfig
git commit -v -s # Create the second commit message
git push --force <your PR branch in your fork>
```

Reply:
Won't that be a problem for contribution guidelines? I thought I was supposed to make only a single squashed commit.

Reply:
> Won't that be a problem for contribution guidelines? I thought I was supposed to make only a single squashed commit.

The guidelines do not say that :) What you should avoid is to add commits that fixes your own commits based on e.g. review comments. You can have multiple commits in a PR when they e.g. change their own thing (like is the case here)

Reply:
Done !